crates/boundless-market/src/contracts/artifacts/IBoundlessMarket.sol

```diff
@@ -246,6 +246,7 @@ interface IBoundlessMarket {
     function lockRequestWithSignature(
         ProofRequest calldata request,
         bytes calldata clientSignature,
+        address prover,
```

Comment:
Related to another comment, we don't need to add this if we don't support smart contract signatures for provers. The `lockRequest` method above supports prover contracts authorizing a lock.
```suggestion
```